activerecord/lib/active_record/connection_adapters/postgresql/oid/type_map_initializer.rb

```diff
@@ -28,9 +28,9 @@ def run(records)
             mapped.each     { |row| register_mapped_type(row)    }
             enums.each      { |row| register_enum_type(row)      }
             domains.each    { |row| register_domain_type(row)    }
-            arrays.each     { |row| register_array_type(row)     }
             ranges.each     { |row| register_range_type(row)     }
             composites.each { |row| register_composite_type(row) }
+            arrays.each     { |row| register_array_type(row)     }
```

Comment:
> Arrays of any built-in or user-defined base type, enum type, composite type, range type, or domain can be created.
https://www.postgresql.org/docs/current/arrays.html

Should call `register_array_type(row)` at last to avoid skipping `register(oid)` while calling https://github.com/rails/rails/blob/beed933b9bb9f317697a95a36da067a67797e0d8/activerecord/lib/active_record/connection_adapters/postgresql/oid/type_map_initializer.rb#L109-L115

Reply:
After https://github.com/rails/rails/pull/40876 merged, PostgreSQLAdapter will run 3 batches of type map related records.

When loading type map records from cache (`initializer.run(self.additional_type_records_cache)`), it will run all the records in the same batch.

Then the order of running `register_xxxx_type` matters.